datafusion/datasource-parquet/src/opener.rs

```diff
@@ -85,7 +86,47 @@ pub(super) struct ParquetOpener {
 }
 
 impl FileOpener for ParquetOpener {
-    fn open(&self, file_meta: FileMeta) -> Result<FileOpenFuture> {
+    fn open(&self, file_meta: FileMeta, file: PartitionedFile) -> Result<FileOpenFuture> {
+        let predicate_creation_errors = MetricBuilder::new(&self.metrics)
+            .global_counter("num_predicate_creation_errors");
+
+        // Prune this file using the file level statistics.
+        // Since dynamic filters may have been updated since planning it is possible that we are able
+        // to prune files now that we couldn't prune at planning time.
+        if let (Some(stats), Some(predicate)) = (&file.statistics, &self.predicate) {
+            let pruning_predicate = build_pruning_predicate(
+                Arc::clone(predicate),
+                &self.table_schema,
```

Comment:
Is it reasonable to use `table_schema` here?

Reply:
It's the only schema we have. And it's not even really the table schema, the name is misleading for historical reasons.

Reply:
It'd be better to add some notes about it. (I often confused when I reading the parquet part code, all kinds of schema, lol)

Reply:
https://github.com/apache/datafusion/blob/46076431359b4b595d00a0a361b083fa25b0188f/datafusion/datasource-parquet/src/opener.rs#L182-L185 😄